src/http.cc

```diff
@@ -833,11 +846,112 @@ HttpStateData::handle1xx(HttpReply *reply)
     // for similar reasons without a 1xx response.
 }
 
+/// if possible, safely ignores the received 1xx control message
+/// otherwise, terminates the server connection
+void
+HttpStateData::drop1xx(const char *reason)
+{
+    if (flags.serverSwitchedProtocols) {
+        debugs(11, 2, "bad 101 because " << reason);
+        const auto err = new ErrorState(ERR_INVALID_RESP, Http::scBadGateway, request.getRaw(), fwd->al);
+        fwd->fail(err);
+        closeServer();
+        // XXX: mustStop("prohibited HTTP/101 response")?
```

Comment:
It depends. 
My sense is that it is not needed, because we are closed the connection.
But produces an easier to follow error in cache.log for debugging.

Reply:
> It should not needed

_Why_ is mustStop() optional here? closeServer() does not trigger a close handler callback -- it is a clean/silent closure IIRC. What check/condition makes mustStop() optional, allowing this job to terminate? I suspect it may have "worked" in your tests because Client::doneAll() would return true after to-server connection closure in your specific tests, but it might still return false in other situations, right?

> all cases we are calling closeServer() inside http.cc it is followed by a mustStop call

Which is one of the motivations for asking this question...

If I do not hear otherwise from you, I will add that mustStop() call.

Reply:
Done in branch commit 03ab2db. I am resolving this change request.